### PR TITLE
Reconnecting to server / host

### PR DESCRIPTION
**Pull request by Ryan089**
## Summary

Allow players to reconnect after disconnecting or failing initial connection attempt.

## Changes to Files

- Lobby updated with game system prefab sceneId's
- Startup updated with online and offline scenes
- Assets updated to unload previous data before loading new data.
- ServerConnectionView updated to actually attempt a reconnection when prompted.

## Proposed validation method
- Create a build of the project in the correct area.
- Go to Edit > Project Settings > SS3D > Application Settings. Set network type to "Client".
- Enter play mode. Scene will attempt to connect, but after a short time will say "connection to server failed".
- Press retry button. Scene will again attempt to connect, but after a short time will say "connection to server failed". 
- Execute the batch file to run the host (SS3D > Builds > Start SS3D Host.bat). Wait until host is in lobby.
- On client, click Retry. Client will enter lobby.
- On client, set as "Ready" and start the round.
- Terminate the host application. Client will be booted, attempt to reconnect, but after a short time will say "connection to server failed".
- Execute the batch file to run the host (SS3D > Builds > Start SS3D Host.bat). Wait until host is in lobby.
- On client, click Retry. Client will enter lobby.

### CODE REVIEW

File: Assets/Content/WorldObjects/Entities/Silicon/EngineeringBorg/EngineeringBorg.prefab [hunk 1]
```diff
@@ -1402,7 +1402,7 @@ MonoBehaviour:
   _isGlobal: 0
   _defaultDespawnType: 0
   NetworkObserver: {fileID: 0}
-  <PrefabId>k__BackingField: 32
+  <PrefabId>k__BackingField: 46
```
Comment (stilnat) [inline Assets/Content/WorldObjects/Entities/Silicon/EngineeringBorg/EngineeringBorg.prefab]: what's that change for ?

Comment (Ryan089) [reply Assets/Content/WorldObjects/Entities/Silicon/EngineeringBorg/EngineeringBorg.prefab]: @stilnat Accidental commit that I must have missed. I have now reverted it.

**Review by stilnat — COMMENTED** [review]
Got an issue with "connecting to server" turning indefinitely when trying to reconnect the second time. That would be failure to go through step 4 of your proposed validation method.
image

**Comment by Ryan089** [comment]
That is bizarre, that was the behaviour before my changes. Not meaning to insult your intelligence, but you are definitely on the right branch, yeah? I just switched back to it on mine and it is definitely working there...

**Comment by stilnat** [comment]
No offense taken, I do that way too much. Yeah I did that. I'm on the good branch now, I created a loop from Hell. See the video on Discord. That happens when you connect as client to host while host is not completely loaded.

**Review by stilnat — CHANGES_REQUESTED** [review]
Need to solve the bug where when the client connects too early to the host, it breaks both client and host and remove their ability to embark.

**Comment by Ryan089** [comment]
Could not resolve bug identified within the review (or even reliably reproduce it - sometimes the bug exhibited different behaviour). At this stage, appears to be beyond my ability to resolve in the short term. Will play around with it locally to see if I can resolve and resubmit.

**Pull request closed without merging**